bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -306,22 +351,68 @@ impl Client {
             let sliding_sync_proxy =
                 self.discovered_sliding_sync_proxy().map(|url| url.to_string());
 
-            Ok(Session {
-                access_token,
-                refresh_token,
-                user_id: user_id.to_string(),
-                device_id: device_id.to_string(),
-                homeserver_url,
-                sliding_sync_proxy,
-            })
+            match auth_api {
+                AuthApi::Matrix(a) => {
+                    let matrix_sdk::matrix_auth::Session {
+                        meta: matrix_sdk::SessionMeta { user_id, device_id },
+                        tokens:
+                            matrix_sdk::matrix_auth::SessionTokens { access_token, refresh_token },
+                    } = a.session().context("Missing session")?;
+
+                    Ok(Session {
+                        access_token,
+                        refresh_token,
+                        user_id: user_id.to_string(),
+                        device_id: device_id.to_string(),
+                        homeserver_url,
+                        oidc_data: None,
+                        sliding_sync_proxy,
+                    })
+                }
+                AuthApi::Oidc(api) => {
+                    let matrix_sdk::oidc::UserSession {
+                        meta: matrix_sdk::SessionMeta { user_id, device_id },
+                        tokens:
+                            matrix_sdk::oidc::SessionTokens {
+                                access_token,
+                                refresh_token,
+                                latest_id_token,
+                            },
+                        issuer_info,
+                    } = api.user_session().context("Missing session")?;
+                    let client_id = api
+                        .client_credentials()
+                        .context("OIDC client credentials are missing.")?
+                        .client_id()
+                        .to_owned();
+                    let client_metadata =
+                        api.client_metadata().context("OIDC client metadata is missing.")?.clone();
+                    let oidc_data = OidcSessionData {
+                        client_id,
+                        client_metadata,
+                        latest_id_token: latest_id_token.map(|t| t.to_string()),
+                        issuer_info,
+                    };
+
+                    let oidc_data = serde_json::to_string(&oidc_data).ok();
+                    Ok(Session {
+                        access_token,
+                        refresh_token,
+                        user_id: user_id.to_string(),
+                        device_id: device_id.to_string(),
+                        homeserver_url,
+                        oidc_data,
+                        sliding_sync_proxy,
+                    })
+                }
+                _ => Err(anyhow!("Unknown authentication API").into()),
```

Comment:
I genuinely don't understand what the catch all case is catching in this instance?

Reply:
Ok right we have that in Swift which I assumed it what it was. But how come it isn't needed here: https://github.com/matrix-org/matrix-rust-sdk/blob/d077892ad431e974487a65d79a9a0fd03e037617/crates/matrix-sdk/src/client/mod.rs#L489-L493

Reply:
Ohhhhh, no I see now. That one is mapping the internal format. I get it now.